sdk/sdk-api/src/main/java/mil/emp3/api/utils/kml/ZipUtils.java

```diff
@@ -0,0 +1,87 @@
+package mil.emp3.api.utils.kml;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+public class ZipUtils {
+
+    private List <String> fileList;
+    private static String OUTPUT_ZIP_FILE = "Folder.zip";
+    private static String SOURCE_FOLDER = "D:\\Reports"; // SourceFolder path
+
+    public ZipUtils() {
+        fileList = new ArrayList < String > ();
+    }
+
+    public static void callZip(String sourceFolder, String outputFolder) {
+        OUTPUT_ZIP_FILE = outputFolder;
+        SOURCE_FOLDER = sourceFolder;
+        ZipUtils appZip = new ZipUtils();
+        appZip.generateFileList(new File(SOURCE_FOLDER));
+        appZip.zipIt(OUTPUT_ZIP_FILE);
+    }
+
+    public void zipIt(String zipFile) {
+        byte[] buffer = new byte[1024];
+        String source = new File(SOURCE_FOLDER).getName();
+        FileOutputStream fos = null;
+        ZipOutputStream zos = null;
+        try {
+            fos = new FileOutputStream(zipFile);
+            zos = new ZipOutputStream(fos);
+
+            System.out.println("Output to Zip : " + zipFile);
+            FileInputStream in = null;
+
+            for (String file: this.fileList) {
+                System.out.println("File Added : " + file);
+                ZipEntry ze = new ZipEntry(source + File.separator + file);
+                zos.putNextEntry(ze);
+                try {
+                    in = new FileInputStream(SOURCE_FOLDER + File.separator + file);
+                    int len;
+                    while ((len = in .read(buffer)) > 0) {
+                        zos.write(buffer, 0, len);
+                    }
+                } finally {
+                    in.close();
+                }
+            }
+
+            zos.closeEntry();
+            System.out.println("Folder successfully compressed");
+
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        } finally {
+            try {
+                zos.close();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    public void generateFileList(File node) {
+        // add file only
+        if (node.isFile()) {
+            fileList.add(generateZipEntry(node.toString()));
+        }
+
+        if (node.isDirectory()) {
+            String[] subNote = node.list();
+            for (String filename: subNote) {
+                generateFileList(new File(node, filename));
+            }
+        }
+    }
+
+    private String generateZipEntry(String file) {
```

Comment:
in general you should put final on your parameters (especially if they are not changing ) throughout your code